### ISSUE TITLE
Drop async_trait in favor of native async traits

### ISSUE THREAD

**Issue by olanod** [paritytech/jsonrpsee#1229]
[`async_fn_in_trait`](https://github.com/rust-lang/rust/pull/115822) has been stabilized, declaring the crate's traits as "native" async traits can bring some performance benefits or make the code simpler, it could also help to towards a no-alloc version of the crate that works better in embedded devices as futures no longer have to be `Box`ed.

**Comment by niklasad1** [comment]
Yes, we are very exited about but it's not released in rustc stable yet and last heard it should be in rustc release in December.
Once it's in rustc stable we will get rid of it.

Would also be nice to get rid off the extra async_trait annotations as well as the Boxed futures.

**Comment by olanod** [comment]
I think it's worth introducing the the change early as code that uses nightly can already benefit(e.g. runtimes) and the feature will likely be in stable by the time a relevant PR gets merged.

**Comment by bkchr** [comment]
> I think it's worth introducing the the change early as code that uses nightly can already benefit(e.g. runtimes) and the feature will likely be in stable by the time a relevant PR gets merged.

We don't use nightly anymore for runtimes and we also don't use async in the runtime. So, no need to "rush". It will come with rust 1.75.0 at 28.12.2023.

**Comment by zancas** [comment]
I think that stable rust is past 1.75 now?
